config/src/config/consensus_config.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    config::{PersistableConfig, SafetyRulesBackend, SafetyRulesConfig},
+    keys::ConsensusKeyPair,
+    trusted_peers::ConsensusPeersConfig,
+};
+use failure::prelude::*;
+use serde::{Deserialize, Serialize};
+use std::path::{Path, PathBuf};
+
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Clone))]
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(default)]
+pub struct ConsensusConfig {
+    pub max_block_size: u64,
+    pub proposer_type: ConsensusProposerType,
+    pub contiguous_rounds: u32,
+    pub max_pruned_blocks_in_mem: Option<u64>,
```

Comment:
can't all of these `Option<T>` be converted to `T` and given a default value in `load()` if not set?

Reply:
that's a good point, there's really no point of having options in the code if we have a default config which makes things much more explicit.... on the flip side it would make an explicit binding to a config instead of having a default locally.

So you could imagine for an integration test, I would need to pull in ConsensusConfig to get the default value of some specific parameter, though that might be a better behavior.